src/cargo/core/source_id.rs

```diff
@@ -393,6 +393,9 @@ impl SourceId {
                     .url
                     .to_file_path()
                     .expect("path sources cannot be remote");
+                if crate::util::toml::is_embedded(&path) {
+                    anyhow::bail!("The embedded script cannot be used as a dependency");
```

Comment:
"embedded script" isn't quite the right term to use for this.

We refer to the manifest as "embedded" because it is encapsulated within the script.  We have unofficially been referring to the files as scripts but I'm not fully satisfied with that term.  In the RFC, I also called these "single file packages".

Reply:
Updated.